app/controllers/tmdb_controller.rb

```diff
@@ -136,13 +140,13 @@ def director_search
     end
   end
 
+  # advanced search
   def discover_search
-    form_params = %i[sort_by date genre actor actor2 company mpaa_rating year_select page]
+    form_params = %i[sort_by date genre actor actor2 company mpaa_rating year year_select page]
     passed_params = params.slice(*form_params).select { |_k, v| v.present? }
     return if passed_params.blank?
 
     searchable_params = SearchParamParser.parse_movie_params(passed_params)
-    tmdb_handler_discover_search(searchable_params)
-    @params_for_view = SearchParamParser.parse_movie_params_for_display(passed_params)
```

Comment:
I moved this into a couple of different places:
* the original search params are returned in the `@results.original_search` so it can be accessed in the view
* the logic to build that big string is moved from the `SearchParamParser` into the `MovieHelper` since it is display/view logic